contrib/tester-progs/fork-tester.c

```diff
@@ -29,10 +29,20 @@ void child1()
 		perror("fork");
 		exit(1);
 	} else if (pid == 0) {
+		int manager_pid = 1; // systemd init PID
 		for (int i = 0;; i++) {
 			int ppid = getppid();
-			if (ppid == 1) {
+			if (ppid == manager_pid) {
 				break;
+			} else {
+				// If systemd run with --user
+				const char *manager_pid_str = getenv("MANAGERPID");
+				if (manager_pid_str) {
+					manager_pid = atoi(manager_pid_str);
+					if (ppid == manager_pid)
+						break;
+					manager_pid = 1;
+				}
```

Comment:
It's not that the forker-tester is a performance-critical program, but reading the environment variable on every iteration seems strange. Is there a reason for not reading it just once? You could also probably avoid the NULL check and just check whether atoi returns 0 and, if it does, just use pid=1.

Reply:
You are right, I should not push code that late. Sorry, I hope the fix will be good enough